src/main/scala/scala/xml/parsing/MarkupParserCommon.scala

```diff
@@ -247,7 +247,7 @@ private[scala] trait MarkupParserCommon extends TokenTests {
       while (true) {
         if (ch == head && peek(rest))
           return handler(positioner(), sb.toString)
-        else if (ch == SU)
+        else if (ch == SU || eof)
           truncatedError("") // throws TruncatedXMLControl in compiler
 
```

Comment:
@ashawley Thank you for the review and comment! It is significant to add this `eof`. This is necessary in test of CDATA section `<broken><![CDATA[A`. Without it, the following OOM occur:

```
Exception in thread "XMLEventReader" java.lang.OutOfMemoryError: Java heap space
        at java.util.Arrays.copyOf(Arrays.java:3332)
        at java.lang.AbstractStringBuilder.expandCapacity(AbstractStringBuilder.java:137)
        at java.lang.AbstractStringBuilder.ensureCapacityInternal(AbstractStringBuilder.java:121)
        at java.lang.AbstractStringBuilder.append(AbstractStringBuilder.java:622)
        at java.lang.StringBuilder.append(StringBuilder.java:202)
        at scala.collection.mutable.StringBuilder.append(StringBuilder.scala:267)
        at scala.xml.parsing.MarkupParserCommon$class.xTakeUntil(MarkupParserCommon.scala:253)
        at scala.xml.pull.XMLEventReader$Parser.xTakeUntil(XMLEventReader.scala:60)
        at scala.xml.parsing.MarkupParser$class.xCharData(MarkupParser.scala:379)
        at scala.xml.pull.XMLEventReader$Parser.xCharData(XMLEventReader.scala:60)
        at scala.xml.parsing.MarkupParser$class.content1(MarkupParser.scala:424)
        at scala.xml.pull.XMLEventReader$Parser.content1(XMLEventReader.scala:60)
        at scala.xml.parsing.MarkupParser$class.content(MarkupParser.scala:459)
        at scala.xml.pull.XMLEventReader$Parser.content(XMLEventReader.scala:60)
        at scala.xml.parsing.MarkupParser$class.element1(MarkupParser.scala:588)
        at scala.xml.pull.XMLEventReader$Parser.element1(XMLEventReader.scala:60)
        at scala.xml.parsing.MarkupParser$class.content1(MarkupParser.scala:433)
        at scala.xml.pull.XMLEventReader$Parser.content1(XMLEventReader.scala:60)
        at scala.xml.parsing.MarkupParser$class.document(MarkupParser.scala:247)
        at scala.xml.pull.XMLEventReader$Parser.document(XMLEventReader.scala:60)
        at scala.xml.pull.XMLEventReader$Parser$$anonfun$run$1.apply(XMLEventReader.scala:96)
        at scala.xml.pull.XMLEventReader$Parser$$anonfun$run$1.apply(XMLEventReader.scala:96)
        at scala.xml.pull.ProducerConsumerIterator$class.interruptibly(XMLEventReader.scala:125)
        at scala.xml.pull.XMLEventReader.interruptibly(XMLEventReader.scala:27)
        at scala.xml.pull.XMLEventReader$Parser.run(XMLEventReader.scala:96)
        at java.lang.Thread.run(Thread.java:745)
```

It turned out that error checking is rudimentary, however, you are quite right. I fix some tests.